### ISSUE TITLE
Document Dynamic Filter

### ISSUE THREAD

**Issue by fuyufjh** [risingwavelabs/risingwave-docs-legacy#493]
Design Docs:

https://github.com/risingwavelabs/rfcs/blob/main/rfcs/0033-dynamic-filter.md

In a nutshell, general non-equal joins are not supported in streaming, except for one special case. if a query with non-equal conditions meets the conditions below:

- The inner side always contains exactly one row. 
- None of the columns from the inner side is required to output.
- The filter condition can be incrementally evaluated when the inner row changed 

See [the design doc](https://github.com/risingwavelabs/rfcs/blob/main/rfcs/0033-dynamic-filter.md) for example.

--------

Additionally, it's worth mentioning a special use case - using a one-row table as a **variable**. For example, `t_erase_ts` is a table to save the minimal timestamp of querying data:

| erase_ts   |
|------------|
| 1673241158 |

Then the following materialized view shows all the data above that timestamp:

```sql
CREATE MATERIALIZED VIEW data_not_erased AS
SELECT t1.* FROM t1 JOIN t_erase_ts
WHERE t1.k > erase_ts

ERROR:  QueryError: Feature is not yet implemented: stream nested-loop join
```
OR

```sql
CREATE MATERIALIZED VIEW data_not_erased AS
SELECT t1.* FROM t1
WHERE t1.value >= (select erase_ts from t_erase_ts);

ERROR:  QueryError: internal error: Scalar subquery might produce more than one row.
```

However, thoese queries will be rejected because RisingWave doesn't know there is always only one row in table `t_erase_ts`. To make it work, simply add an aggregation there:

```sql
CREATE MATERIALIZED VIEW data_not_erased AS
SELECT t1.* FROM t1
WHERE t1.k > (select MAX(erase_ts) from t_erase_ts)
```

Then, Dynamic Filter will be adopted to run this streaming query.

```
 StreamMaterialize { columns: [k, value], pk_columns: [k] }
 └─StreamDynamicFilter { predicate: (t1.value >= max(max(t_erase_ts.erase_ts))) }
   ├─StreamTableScan { table: t1, columns: [k, value] }
   └─StreamExchange { dist: Broadcast }
     └─StreamProject { exprs: [max(max(t_erase_ts.erase_ts))] }
       └─StreamGlobalSimpleAgg { aggs: [sum0(count), max(max(t_erase_ts.erase_ts))] }
         └─StreamExchange { dist: Single }
           └─StreamHashAgg { group_key: [Vnode(t_erase_ts._row_id)], aggs: [count, max(t_erase_ts.erase_ts)] }
             └─StreamProject { exprs: [t_erase_ts.erase_ts, t_erase_ts._row_id, Vnode(t_erase_ts._row_id)] }
               └─StreamTableScan { table: t_erase_ts, columns: [erase_ts, _row_id] }
```

**Comment by fuyufjh** [comment]
@hengm3467 Can you help to assign it?

**Comment by hengm3467** [comment]
Closed as the PR #505 was merged. @fuyufjh Please let me know if you have additional comments or suggestions about the topic: https://www.risingwave.dev/docs/upcoming/sql-pattern-dynamic-filter/
